extensions/mssql/src/objectManagement/ui/S3AddBackupFileDialog.ts

```diff
@@ -49,7 +49,8 @@ export class S3AddBackupFileDialog extends DialogBase<S3AddBackupFileDialogResul
 		this.connectionUri = connectionUri;
 
 		this.dialogObject.okButton.onClick(async () => {
-			this.result.backupFilePath = `s3://${this.bucketDropdown.value}.s3.${this.regionInputBox.value}.amazonaws.com/${this.backupFilesDropdown.value}`;
+			// s3 objects can contain special characters that may cause issues with http request
+			this.result.backupFilePath = `s3://${this.bucketDropdown.value}.s3.${this.regionInputBox.value}.amazonaws.com/${encodeURIComponent(this.backupFilesDropdown.value.toString())}`;
```

Comment:
Would it be easier to run encodeURI on the whole string like with `result.s3Url` below?